Android/app/src/main/java/app/intra/net/go/GoIntraListener.java

```diff
@@ -62,13 +62,13 @@ public void onTCPSocketClosed(TCPSocketSummary summary) {
     // - DURATION: socket lifetime in seconds
     // - TODO: FIRST_BYTE_MS: Time between socket open and first byte from server, in milliseconds.
 
-    Bundle bytesEvent = new Bundle();
-    bytesEvent.putLong(Names.UPLOAD.name(), summary.getUploadBytes());
-    bytesEvent.putLong(Names.DOWNLOAD.name(), summary.getDownloadBytes());
-    bytesEvent.putInt(Names.PORT.name(), summary.getServerPort());
-    bytesEvent.putInt(Names.TCP_HANDSHAKE_MS.name(), summary.getSynack());
-    bytesEvent.putInt(Names.DURATION.name(), summary.getDuration());
-    FirebaseAnalytics.getInstance(vpnService).logEvent(Names.BYTES.name(), bytesEvent);
+    new AnalyticsEvent(vpnService)
+        .put(Params.UPLOAD, summary.getUploadBytes())
```

Comment:
I'm wondering if it makes sense to create factory functions or static methods like SendBytesEvent(upload_bytes, download_bytes, port, tcpHandshakeMs, duration)

That would hide all the Params details from the application code and provide some type safety. Also, most of the time you set the params and send in the same call.

Even though it may result in a little more code, you'll be isolating complexity and the application code will be more readable and simpler (e.g. 1 line instead of 7 to send an event).

Reply:
Done!